### PR TITLE
instrumentation of Document.referrer and Window.location Web APIs

### PR DESCRIPTION
**Pull request by jueckstock**
Added WebAPI instrumentation points to record read access to `document.referrer` and read/write access to `location.href` and friends (all properties of `window.location` with custom getter/setter bindings and the standard `Location` mutation methods).

Used a new enumeration-based interface for registering Web API activity (like the one already in place for JS built-in APIs); existing Web API instrumentation can/should be migrated to this new interface in the future as time permits.

### CODE REVIEW

File: patches/third_party-blink-renderer-core-dom-document.cc.patch [hunk 1]
```diff
@@ -1,5 +1,5 @@
 diff --git a/third_party/blink/renderer/core/dom/document.cc b/third_party/blink/renderer/core/dom/document.cc
-index 02e6831f00c14a9aab6ba6d4218a94ec133c6076..091eb1776f2ae37992024de1e641ed64b3becbe9 100644
+index 2247f71c3c97..7eaca00dc6f2 100644
```
Comment (antonok-edm) [inline patches/third_party-blink-renderer-core-dom-document.cc.patch:2]: You should be able to revert the changes to the headers and line numbers here safely as well (everything up to line 228)

Comment (jueckstock) [reply patches/third_party-blink-renderer-core-dom-document.cc.patch]: done
